src/playlist.js

```diff
@@ -216,21 +216,34 @@ export const sumDurations = function(playlist, startIndex, endIndex) {
  * window which is the duration of the last segment plus 2 target durations from the end
  * of the playlist.
  *
+ * A liveEdgePadding can be provided which will be used instead of calculating the safe live edge.
+ * This corresponds to suggestedPresentationDelay in DASH manifests.
+ *
  * @param {Object} playlist
  *        a media playlist object
+ * @param {Number} liveEdgePadding
+ *        A number in seconds indicating how far from the end we want to be.
+ *        If provided, this value is used instead of calculating the safe live index from the target durations.
+ *        Corresponds to suggestedPresentationDelay in DASH manifests.
  * @return {Number}
  *         The media index of the segment at the safe live point. 0 if there is no "safe"
  *         point.
  * @function safeLiveIndex
  */
-export const safeLiveIndex = function(playlist) {
+export const safeLiveIndex = function(playlist, liveEdgePadding) {
   if (!playlist.segments.length) {
     return 0;
   }
 
-  let i = playlist.segments.length - 1;
-  let distanceFromEnd = playlist.segments[i].duration || playlist.targetDuration;
-  const safeDistance = distanceFromEnd + playlist.targetDuration * 2;
+  let i = playlist.segments.length;
+  let lastSegmentDuration = playlist.segments[i - 1].duration || playlist.targetDuration;
+  const safeDistance = typeof liveEdgePadding === 'number' ? liveEdgePadding : lastSegmentDuration + playlist.targetDuration * 2;
+
+  if (safeDistance === 0) {
```

Comment:
Do we want this to account for any distance less than last segment duration?

Reply:
no, I think we should only special-case 0 exactly. Any other value should be ran through the loop below to find the appropriate return value.